syntaxes/renpy.tmLanguage_2.json

```
@@ -0,0 +1,165 @@
+{
+  "name": "Ren'Py",
+  "scopeName": "source.renpy",
+  "firstLineMatch": "^#!/.*\\brenpy\\b",
+  "patterns": [
+    { "include": "#comments" },
+    { "include": "#python-statements" },
+    { "include": "#expressions" }
+  ],
+  "repository": {
+    "python-statements": {
+      "name": "embedded.python",
+      "patterns": [
+        {
+          "comments": "Renpy python block",
+          "name": "embedded.python.block",
+          "begin": "^(init\\s-?\\d?\\s?)?(python:)",
```

Only catches single digit init levels (`-9` to `9`) without any storemodules. This should fix that issue
Also catches `python early` too

```suggestion
          "begin": "^(init\\s-?\\d*\\s?)?(python(\\searly)?(\\sin\\s\\w+)?:)",
```

I'm not too sure how to make this highlight the storemodule name, init level, or `early` keyword

---

I just realized another mistake of mine. This regular expression only matches when the `python` block is at the beginning of the line, i.e. there is no indentation. But if we fix it now the grammar won't know where a python block ends. We have to use the semantic token provider to know when a python block ends, based on its indentation.

---

You're right! If we do this semantically too, we may be able to fix breadcrumbs while we're at it. I've noticed the current code treats everything as 0 length scopes, so breadcrums never show because nothing is really in a scope, as all "scopes" have no length

---

Can you test it again?